migrations/0021-add-userFeedback-table.ts

```diff
@@ -0,0 +1,64 @@
+import { MigrationInterface, QueryRunner, Table } from 'typeorm';
+
+export class AddFeedbackTable1602394486221 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.createTable(new Table({
+      name: 'Feedback',
+      columns: [
+        {
+          name: 'id',
+          type: 'integer',
+          isGenerated: true,
+        },
+        {
+          name: 'uuid',
+          type: 'uuid',
+          isPrimary: true,
+          isGenerated: true,
+          generationStrategy: 'uuid',
+        },
+        {
+          name: 'user',
+          type: 'uuid',
+        },
+        {
+          name: 'title',
+          type: 'varchar(255)',
+        },
+        {
+          name: 'description',
+          type: 'text',
+          isNullable: true,
```

Comment:
Should this be nullable?

Reply:
I figured a description doesn't _need_ to be required by the user if they're content with just the title. Should this be `NOT NULL` and just store empty string descriptions if none is submitted?

Reply:
I think we should encourage some detail, but I guess as usual, ask design. Do we also want to be able to delete feedback/mark as spam or anything like that?

Reply:
I think so in case people try to write noisy feedback to potentially gain points. Maybe a `hidden` field on feedback to hide it on the admin side but not delete it entirely?

Reply:
Tossup on if we want to hide or delete. On one hand, it's just spam so it can be deleted, but on the other someone could hide feedback they don't want others to see (although that's bad faith and someone could just as easily edit the database). Implementation-wise, it'd be simpler to just delete instead of a `markedAsSpam` field, but it could go either way. Ask design I suppose?

Should we email all feedback somewhere, so someone's actually looking at it? Although they'd still have to mark it as acknowledged in the portal. Also is some feedback worth more than others?

Reply:
A person who submitted feedback would be able to view all their submitted feedback, so with deleting feedback from the admin side, doesn't that also delete it for the person's view too? Like they would all of a sudden not see one of their feedback they submitted without being notified of that. I'll ask design about that

Hmm I'm not sure about emailing feedback somewhere, because we could potentially have hundreds of feedback per week from members so emailing all those would be extra, plus if its in the portal, admin can read + acknowledge in one place.

re: some feedback more than others - design has it where everyone gets 5 right now (not even on acknowledgement, just on submission, which should probably be moved to acknowledgement), and I think that's fine, like if the feedback passes the threshold of acknowledgement anyway then it deserves 5 points.